gp-res-filter/src/test/java/com/ibm/g11n/pipeline/resfilter/impl/GlobalizeJsResourceTest.java

```diff
@@ -57,67 +63,61 @@
+        lst.add(ResourceString.with("$.Colors.Blue", "B").sequenceNumber(6).build());
+        lst.add(ResourceString.with("Done", "Finish").sequenceNumber(7).build());
 
         Collections.sort(lst, new ResourceStringComparator());
         EXPECTED_INPUT_RES_LIST = lst;
     }
 
-    private static Bundle WRITE_BUNDLE;
+    private static LanguageBundle WRITE_BUNDLE;
 
     static {
-        WRITE_BUNDLE = new Bundle();
-        WRITE_BUNDLE.addResourceString("Action", "Aktion", 1);
-        WRITE_BUNDLE.addResourceString("File", "Datei", 2);
-        WRITE_BUNDLE.addResourceString("longText", LONG_TEXT_STRING_DE, 3);
-        WRITE_BUNDLE.addResourceString("$.Colors.Red", "Rot", 4);
-        WRITE_BUNDLE.addResourceString("$.Colors.Green", "Gr\u00FCn", 5);
-        WRITE_BUNDLE.addResourceString("$.Colors.Blue", "Blau", 6);
-        WRITE_BUNDLE.addResourceString("Done", "Fertig", 7);
+        LanguageBundleBuilder bundleBuilder = new LanguageBundleBuilder(false);
+        bundleBuilder.addResourceString("Action", "Aktion", 1);
+        bundleBuilder.addResourceString("File", "Datei", 2);
+        bundleBuilder.addResourceString("longText", LONG_TEXT_STRING_DE, 3);
+        bundleBuilder.addResourceString("$.Colors.Red", "Rot", 4);
+        bundleBuilder.addResourceString("$.Colors.Green", "Gr\u00FCn", 5);
+        bundleBuilder.addResourceString("$.Colors.Blue", "Blau", 6);
+        bundleBuilder.addResourceString("Done", "Fertig", 7);
+        bundleBuilder.embeddedLanguageCode("de");
+        WRITE_BUNDLE = bundleBuilder.build();
     }
 
     private static final GlobalizeJsResource res = new GlobalizeJsResource();
 
     @Test
-    public void testParse() throws IOException {
+    public void testParse() throws IOException, ResourceFilterException {
         assertTrue("The input test file <" + INPUT_FILE + "> does not exist.", INPUT_FILE.exists());
 
         try (InputStream is = new FileInputStream(INPUT_FILE)) {
-            Bundle bundle = res.parse(is);
+            LanguageBundle bundle = res.parse(is, null);
             List<ResourceString> resStrList = new ArrayList<>(bundle.getResourceStrings());
             Collections.sort(resStrList, new ResourceStringComparator());
             assertEquals("ResourceStrings did not match.", EXPECTED_INPUT_RES_LIST, resStrList);
         }
     }
 
     @Test
-    public void testWrite() throws IOException {
+    public void testWrite() throws IOException, ResourceFilterException {
         File tempFile = File.createTempFile(this.getClass().getSimpleName(), ".json");
         tempFile.deleteOnExit();
 
         try (OutputStream os = new FileOutputStream(tempFile)) {
-            res.write(os, "de", WRITE_BUNDLE);
+            res.write(os, WRITE_BUNDLE, new FilterOptions(Locale.GERMAN));
             os.flush();
             assertTrue(ResourceTestUtil.compareFiles(EXPECTED_WRITE_FILE, tempFile));
         }
     }
 
+ // TODO: Not ready yet
```

Comment:
TODO - does this need a new issue?

Reply:
"merge" is optional operation, and if not implemented, it's same as "write". GlobalizeJsResource does not have implementation. But we "may" implement it later time.  I just added a comment - to be addressed later.